mantidproject/mantid
# Scd445 integrate ellipsoids wrapper

**Description of work.**

This PR reintroduces the last version of `IntegrateEllipsoids` as V1 and labels the latest version as V2.  In addition, it also introduces a wrapper that will automatically detect and switch between the two based on input.

The reason for this change is the latest version of `IntegrateEllipsoids` requires much larger memory, as well as computation time in exchange for being able to address the use case for unindexed satellite peaks, this split allows for cases in which unindexed satellite peaks arent a concern to save memory and compute time.

**To test:**
pass either `IntegrateInHKL` or `GetUBFromPeaksWorkspace` as true, and a peaksworkspace with an index count greater than 0 to `IntegrateEllipsoids` to trigger version 1 to run.(NOTE: SharedBackground must not be true, it is by default false)

Any case that doesnt match the above will trigger V2 instead.

You should be able to observe a difference in logs/memory/computation time.





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
